src/components/stores/messageStore.ts

```diff
@@ -92,7 +94,33 @@ const createTaskSlice: StateCreator<
           tasks: updatedTasks,
         };
       });
+    },  
+    reorderTasks: (startIdx: number, targetIdx: number) => {
+      set((state) => {
+          if (startIdx < 0 || startIdx >= state.tasks.length || targetIdx < 0 || targetIdx >= state.tasks.length){
+            return {...state};
+          }
+          const updatedTasks = Array.from(state.tasks);
+          const [removed] = updatedTasks.splice(startIdx, 1);
+          updatedTasks.splice(targetIdx, 0, removed!);
+
```

Comment:
Quick Question: if line 104 changes the indexing of the elements, would the splice at targetIdx on 105 be still accurate? (e.g. what happens if startIdx < targetIdx?)

Reply:
Good question! 

It seems that the way we get targetIdx (result.destination.index) takes this into account already, because it will know know where we want to place the new task and "shift everything" for us.